components/menu/submenu-title.component.ts

```diff
@@ -96,12 +97,13 @@ export class NzSubMenuTitleComponent implements OnDestroy, OnInit {
   }
 
   setMouseState(state: boolean): void {
-    if (!this.nzDisabled) {
+    if ((!this.nzDisabled && this.nzTriggerSubMenuAction === 'hover') || !state) {
       this.subMenuMouseState.next(state);
     }
   }
   clickTitle(): void {
-    if (this.mode === 'inline' && !this.nzDisabled) {
+    if ((this.mode === 'inline' || this.nzTriggerSubMenuAction === 'click') && !this.nzDisabled) {
+      this.subMenuMouseState.next(true);
```

Comment:
1. It'll be great if support closing submenu when clicking outside of the submenu-title
2. Please check that sub-sub-menu will be closed recursively if closing the higher-level submeu. For example, submenu of Option3 should be closed when closing submenu of Option2
```txt
- Option 1
- Option 2
   - Option 3
      - Option 4
```

Reply:
Good call, both these behavours are now implemented.